draft-ietf-mls-protocol.md

```diff
@@ -2757,21 +2761,34 @@ External Commits.
 
 External Commits work like regular Commits, with a few differences:
 
-* External Commits MUST reference an Add Proposal that adds the issuing new
-  member to the group
+* The proposals included by value in an External Commit MUST meet the following
+  conditions:
+  * There MUST be a single Add proposal that adds the new issuing new member to
+    the group
+  * There MUST be a single ExternalInit proposal
+  * There MUST NOT be any Update proposals
+  * If a Remove proposal is present, then:
```

Comment:
We would need to require that the prior identity (if there was prior membership) is removed. The proposer should do this - so maybe mandating a "must" for including a remove proposal if resyncing. Basically, the group should not end up with duplication of the identity (and potential backdoor). It is also inefficient for healing as Alice needs to replace keys at both nodes if both of the "Alice" identities are allowed to stay.

This is presents a question for handling sub-identities (if Alice has multiple devices and the identity is linked to Alice vs. a device). In that case an Add proposal for extra devices could include a PSK derived from the group, but the group will never know which device Alice is using (since the identity is the same). So what is the assumption here with respect to multi-device handling?

Reply:
I believe PR #476 would allow us to check both user _and_ device identity. But of course this depends on how the application implements multi-device. They could user a different `identity` for each device, or a different `endpoint_id` or both.